src/mode/graphics.rs

```diff
@@ -116,13 +130,65 @@ where
         self.properties
             .set_draw_area((0, 0), (display_width, display_height))?;
 
+        self.min_x = 255;
+        self.max_x = 0;
+        self.min_y = 255;
+        self.max_y = 0;
+
         match display_size {
             DisplaySize::Display128x64 => self.properties.draw(&self.buffer),
             DisplaySize::Display128x32 => self.properties.draw(&self.buffer[0..512]),
             DisplaySize::Display96x16 => self.properties.draw(&self.buffer[0..192]),
         }
     }
 
+    /// Write out data to a display.
+    /// 
+    /// This is typically faster than a regular flush since it only updates the parts of the
+    /// display that have changed since the last flush. 
+    /// 
+    /// This is slower than a regular flush when the size of the updated area approaches the full
+    /// size of the display, so in that case this function simply calls flush.
+    pub fn fast_flush(&mut self) -> Result<(), DI::Error> {
+        if self.max_x < self.min_x || self.max_y < self.min_y {
+            return self.flush();
+        }
+
+        let display_size = self.properties.get_size();
+        let (width, height) = display_size.dimensions();
+        let width = width as usize;
```

Comment:
There's a bit of type fiddling going on here. Removing this line would remove the need for `width as u8` on line 165. The `as usize` could then be moved to the `bounded_draw()` call at the end of the method.